src/device.rs

```diff
@@ -4,12 +4,86 @@
 //
 // SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
 
+use super::queue::Queue;
 use super::*;
-use std::sync::atomic::AtomicUsize;
 use std::sync::Arc;
+use vm_device::interrupt::{InterruptIndex, InterruptSourceGroup, InterruptSourceType};
```

Comment:
We have to think this through. Right now it looks like a dependency on vm-device from vm-virtio will introduce a cyclic dependency between those two. vm-virtio device depends on vm-device because it needs interrupts and vm-device depends on vm-virtio because the DeviceManager manages devices including VirtioDevices.

What are your thoughts on this?

Reply:
As @sameo  mentioned, the vm-virtio has dependency on the vm-device crate, but not vice versa.

Reply:
Ok, I'll look again at the vm-device crate. What PR should I look at to understand the design of interrupts?